apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UploadObjectOptions.cs

```diff
@@ -185,6 +195,18 @@ internal void ModifyMediaUpload(CustomMediaUpload upload)
             {
                 upload.Options.ModifySessionInitiationRequest += message => message.Headers.Add("Origin", Origin);
             }
+            if (RetentionMode != null || RetainUntilTime != null)
+            {
+                upload.Body.Retention = new Object.RetentionData();
+                if (RetentionMode != null)
+                {
+                    upload.Body.Retention.Mode = RetentionMode;
+                }
+                if (RetainUntilTime != null)
+                {
+                    upload.Body.Retention.RetainUntilTimeDateTimeOffset = RetainUntilTime;
```

Comment:
Similarly here, we have a pubic overload that acceots the object to upload so users can use that directly to set these.